src/Chapter2.hs

```diff
@@ -857,7 +881,11 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind [] = []
+rewind l =  x : rewind xs
+    where 
+      (x:xs) = rotate (length l -1) l
```

Comment:
I'm afraid, this implementation will be slow. `rotate` works in linear time depending on the list size, rotating the list on each recursive call is very slow...

Reply:
No I get how to `rewind` in linear time, thanks for pointing out.